### ISSUE TITLE
Bug: `sort-classes`: Nested/Optional chained/Non-null asserted/Unary property references are not detected as dependencies

### ISSUE THREAD

**Issue by hugop95** [azat-io/eslint-plugin-perfectionist#243]
### Describe the bug

See code example.

### Code example

```ts
class Class {
  b = new Subject();
  a = this.b;
}
```
is valid: `a` is correctly detected as a dependency of `b`.

However:
```ts
class Class {
  b = new Subject();
  a = this.b.asObservable();
}
```
is not.

### ESLint version

8.57.0

### ESLint Plugin Perfectionist version

v3.3.0

### Additional comments

Also includes optional chained non-null asserted and unary properties.

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Fixed in e7c113d.
Released in [v3.4.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v3.4.0).
